src/backing/client/testing_util/is_proto_equal.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modifications copyright 2020 Google LLC
+ *
+ *    - Renamed namespaces and file includes
+ *    - Replaced Cloud C++ optional implementation with absl::optional
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KMSENGINE_BACKING_CLIENT_TESTING_UTIL_IS_PROTO_EQUAL_H_
+#define KMSENGINE_BACKING_CLIENT_TESTING_UTIL_IS_PROTO_EQUAL_H_
+
+#include <string>
+
+#include <google/protobuf/message.h>
+#include <gmock/gmock.h>
+
+#include "absl/types/optional.h"
+
+namespace kmsengine {
+namespace backing {
+namespace client {
+namespace testing_util {
+
+absl::optional<std::string> CompareProtos(
+    google::protobuf::Message const& arg,
+    google::protobuf::Message const& value);
+
+MATCHER_P(IsProtoEqual, value, "Checks whether protos are equal") {
```

Comment:
I think the third argument in the MATCHER_P would correspond to the output error message, so it should probably be more like:
```
MATCHER_P(EqualsProto, proto, absl::StrFormat("proto %s", negation ? "does not equal" : "equals")) {
  ...
}
```

Also, using a name like `EqualsProto` is more idiomatic. That way you can have
```
ASSERT_THAT(p1, EqualsProto(p2));
```
which flows better in terms of readability.